### DISCUSSION TITLE
open source and free

### DISCUSSION THREAD

**Discussion by dinarpay** [aliyazdi75/flutter_chat#4] [Q&A] [answered]
My friend
Is the app open source and free
What are the features of the application, is it like WhatsApp
Do I need a server and monthly fees to operate the voice and video chat service?

**Comment by aliyazdi75** [comment, marked as answer]
Yes, I believe all applications in the world must be open-source.
I told you, read the readme!
It uses webRTC, first, you should learn what webRTC is.
No, a simple socket server for handshaking would be enough.

**Reply by dinarpay** [reply]
Yes, is the server 1 GB RAM - 1 GB processor
Capable of processing 5000 users, for voice and video chatting

**Reply by aliyazdi75** [reply]
I don't exactly know about those figures, but I would recommend using cloud-based like firebase for that user number.
A STUN server is simple doesn't require high bandwidth but for the TURN server, you should consider a high one.

**Comment by jupli** [comment]
Salam, please tell me more how to run this apps? i already installed but its keep running in page of authentication for e-mail.

**Reply by aliyazdi75** [reply]
Salam my friend,

It doesn't work without the server, but you can make a simple socket server for that and run the application.

**Reply by jupli** [reply]
ya Ali, can you help for making simple socket server? or you can point the source code ?

**Reply by jupli** [reply]
how's in there ? yesterday was great Asyura celebrate in here

**Reply by aliyazdi75** [reply]
Yes just search on google asp.net socket server and make some tables for chat and users based on this project models. It would be amazing if you could create an open-source one and I will add it to this project.

That's nice. Here sucks :(

**Reply by jupli** [reply]
so far i have manage to build the chat system using php, flutter and mysql, now i will try to create socket server, and i hope i will put all the codes into yours github to make it open sources.

**Reply by maytham553** [reply]
> so far i have manage to build the chat system using php, flutter and mysql, now i will try to create socket server, and i hope i will put all the codes into yours github to make it open sources. 

We can work together to make server side if you like .
